### DISCUSSION TITLE
Передача ПВЗ в письма и приложение

### DISCUSSION THREAD

**Discussion by mraznoe** [cdek-it/wordpress#31] [Ideas]
Здравствуйте! 
1. Будет ли реализована передача адреса ПВЗ или адреса курьера в текст письма заказа?
2. Мы используем официальное приложение для работы с магазином на мобильном устройстве https://woocommerce.com/mobile/ и в это приложение адреса тоже не передаются. Получается замкнутый круг. Если заказ с доставкой СДЭК нужно лезть в админку через десктоп. Возможно ли как-то решить передачу адреса доставки СДЭК в письмо и/или в приложение. В других плагинах доставки виден адрес через доп. поля.

**Comment by vermorag** [comment]
Добрый день.
1. В настоящий момент, такого функционала нет в планах, поскольку после регистрации заказа клиенту приходит письмо от СДЭК-а о том, что заказ был зарегистрирован. В письме есть трек номер и адрес получения. Однако, если будет больше обращений по данному вопросу, то доработка может быть включена в план к реализации.
2. Адрес и код ПВЗ, отображающийся администратору на странице информации о заказе сохранен в доп. поля метода доставки заказа. По вопросу отображения этих полей в мобильном приложении необходимо задать вопрос разработчикам данного приложения.

**Reply by Dustor** [reply]
Давно уже никто из покупателей не читает письма, это прошлый век.
Функции просмотра адреса пункта выдачи в e-mail (для админа) на самом деле очень не хватает, почему бы не добавить это в какое-то поле (например, адрес доставки)? Потому что у меня на сайте, к примеру, менеджер всегда предварительно согласовывает с клиентом адрес доставки (клиент может промазать с пунктом выдачи). И для этой информации приходится отдельно залезать в админку на каждый заказ. 

Поэтому я добавил себе в письма-уведомления эту информацию вручную, но это такой себе способ... 
Пример моего кода для admin_new_order.php (я не программист, если что, просто скопипастил код из самого же СДЭК):

namespace Cdek\UI;
use Cdek\CdekApi;
use Cdek\Config;
use Cdek\Helpers\WeightCalc;
use Cdek\MetaKeys;
use Exception;
use WC_Order_Item_Shipping;
try {
    $shipping_method = $order->get_shipping_method();
    $data = $order->get_data(); // данные заказа
    $shipping_methods = $order->get_shipping_methods();
    foreach ($shipping_methods as $shipping_method) {
        $shipping_method_meta_code = $shipping_method->get_meta('_official_cdek_office_code');
        if( !empty($shipping_method_meta_code)) {
            $officeAddress = json_decode((new CdekApi)->getOffices(['code' => $shipping_method_meta_code])['body']
                , true, 512, JSON_THROW_ON_ERROR);
    
            if (empty($officeAddress[0]['location']['address'])) {
            } else {
                echo '<p style="font-size:1.7em;line-height:1.4em;"><b>Доставка в пункт СДЭК:</b><br> ' 
                    . esc_html__( $data['shipping']['state'], 'woocommerce') . ', ' . esc_html__( $data['shipping']['city'], 'woocommerce')  
                    . ', ' . esc_html__( $officeAddress[0]['location']['address'], 'woocommerce') . '</p>';
            }
        }
    }
} catch (Exception $exception) {
    echo '<p>Уточнить доставку</p>';
}

**Reply by Nick-757** [reply]
Читают и еще как. Не однократно покупатели выбирали не тот ПВЗ и звонили просили исправить на другой адрес это с плагином от другого разработчика). Самое простое решение, чтобы в письма попадало, добавлять выбранный ПВЗ в поле billing_address_1 , так делают другие плагины, например Яндекс Доставка (MeaSoft) и 5Post.

**Comment by mraznoe** [comment]
1. Не все предают данные о заказе сразу в СДЭК и многие ведут переписку по заказу в ватсап. После заказа письмо может отправляться и покупателю. А в нем совершенно другой адрес указывается. Если есть возможность сделайте, пожалуйста, верный адрес, чтобы нигде не было путаницы. В личном кабинете покупатель тоже видит неверный адрес. Это путает всех!

2. Специально проверили несколько заказов и в доп. полях нет данных о ПВЗ.
Прикрепляю два скриншота. В одном как выглядит заказ СДЭК, в другом от другой ТК. От другой ТК в приложении показывает ПВЗ в доп полях, от СДЭК - нет.
![1пвз](https://github.com/user-attachments/assets/f12ef89d-c31d-48d3-9569-796d8d608d13)
![2пвз](https://github.com/user-attachments/assets/d159009b-b83a-4d78-962d-64a63a4d8c78)

**Reply by vermorag** [reply]
1. В письме указывается адрес, который выбрал покупатель при совершении заказа и который указан в заказе. Письмо отправляется функционалом вукоммерса на основе данных, которые есть у него. Наш плагин в этом процессе не участвует.
2. На вашем же скриншоте видна информация о ПВЗ с адресом. Она хранится в дополнительных полях доставки. Сохранять ее в поля заказа не планируется, поскольку это свойство доставки, а не заказа. А методов доставки у вукоммерса в заказе может быть несколько.

**Reply by Nick-757** [reply]
> 1. В письме указывается адрес, который выбрал покупатель при совершении заказа и который указан в заказе. Письмо отправляется функционалом вукоммерса на основе данных, которые есть у него. Наш плагин в этом процессе не участвует.
> 2. На вашем же скриншоте видна информация о ПВЗ с адресом. Она хранится в дополнительных полях доставки. Сохранять ее в поля заказа не планируется, поскольку это свойство доставки, а не заказа. А методов доставки у вукоммерса в заказе может быть несколько.

К сожалению в последней версии 5.0.0 плагина, в адрес доставки billing_address_1 не подставляется адрес выбранного ПВЗ. В более поздних версиях было. Пожалуйста верните эту возможность. Покупатель будет видеть адрес выбранного ПВЗ.
